tests/test_torch_tiles.py

```diff
@@ -476,3 +489,109 @@ def set_bm_nm(
     out = linear_torch(inp)
     out_ref = linear(inp)
     assert allclose(out, out_ref, atol=1e-5)
+
+
+@mark.parametrize("dtype", [float32, float16, bfloat16])
+@mark.parametrize("device_type", [device("cpu"), device("cuda")])
+def test_dtype(dtype: torch_dtype, device_type: device):
+    """
+    Test whether there is some implicit casting of the inputs.
+
+    Args:
+        dtype: Torch dtype
+        device_type: Torch device
+
+    Raises:
+        SkipTest: If cuda not found or disabled
+    """
+    if SKIP_CUDA_TESTS or not cuda.is_available():
+        raise SkipTest("CUDA not available")
+
+    rpu_config_torch = TorchInferenceRPUConfig()
+    rpu_config_torch.forward.is_perfect = True
+
+    linear_torch = AnalogLinear(
+        in_features=256, out_features=256, bias=False, rpu_config=rpu_config_torch
+    )
+    linear_torch = linear_torch.to(device=device_type, dtype=dtype)
+    inp = randn((10, 256)).to(device=device_type, dtype=dtype)
+    out: Tensor
+    out = linear_torch(inp)  # pylint: disable=not-callable
+    assert out.dtype == dtype
+
+
+@mark.parametrize("dtype", [float32, float16, bfloat16])
+@mark.parametrize("device_type", [device("cpu"), device("cuda")])
+@mark.parametrize("use_autocast", [True, False])
```

Comment:
While these tests are nice, they fall out of the usual practice in the aihwkit. We actually have a test suite and parameterized test cases in tests/helpers that should be used an extended for uniformity in the aihwkit.

Reply:
For the torch tile, we have some other tests that are in this format because I wanted to test specific parameters. The other option would be to add these parameters as well before the `class TorchInferenceTest(ParametrizedTestCase):`, right?